Makefile

```diff
@@ -18,15 +18,31 @@ escape_double_quote = $(subst $\",$\\",$(1))
 
 .PHONY: all
 
+# Builds LIGO contract. Arguments:
+#   1: The source file
+#   2: The target file
+#   3: Ligo pragmas
+define build_ligo
+	@mkdir -p $(dir $(2))
+
+	@ #Add necessary #define pragmas first
+	$(eval TOTAL_FILE := $(shell mktemp $(1).total-XXX))
+	$(foreach CVAR,$(3),$(file >>$(TOTAL_FILE),#define $(CVAR)))
+	@cat $(1) >> $(TOTAL_FILE)
+
+	# ============== Compiling `$(1)` with options `$(3)` ============== #
+	@$(BUILD) $(TOTAL_FILE) main --output-file $(2)
+	@rm $(TOTAL_FILE)
+endef
+
 all: \
-	$(OUT)/segmented_cfmm.tz
+	$(OUT)/segmented_cfmm_default.tz
 
-# Compile LIGO contract into its michelson representation.
-$(OUT)/segmented_cfmm.tz: ligo/**
-	mkdir -p $(OUT)
-	# ============== Compiling contract ============== #
-	$(BUILD) ligo/main.mligo main --output-file $(OUT)/segmented_cfmm.tz
+$(OUT)/segmented_cfmm_default.tz : LIGO_PRAGMAS = DUMMY_PRAGMA1 DUMMY_PRAGMA2
```

Comment:
Would be nice to see how we use this `PRAGMA` in the code, just like for example: `TOKEN_IS_FA2` in ctez: https://github.com/tezos-checker/ctez/blob/main/cfmm.mligo.

Or do we intend to do this in a separate issues/PR ?

Reply:
Yes, I hope that the pragmas will be set and used properly in the respective PRs that add various token types or allow disabling the protocol fees.

For now I can insert some dummy

```
#if !DUMMY_PRAGMA1
An example of conditionally present code, remove it once normal pragmas are set.
#endif
```

as an example, if you think it is appropriate. Now I think it would be a good idea.

Reply:
Yeah, I'm good with that.